src/vscode-cosmosdb.api.d.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+export interface VSCodeCosmosDB {
+    /**
+     * Returns connection string of the CosmosDB item by specified id. Throw an error if the item doesn't have a connection string.
+     * @param treeItemId The id of the CosmosDB tree item
+     */
+    readonly getConnectionString: (treeItemId: string) => Promise<string>;
+
+    /**
+     * Traverses the CosmosDB tree with a quick pick at each level. Goes until find item with database-level context value. Returns the CosmosDB tree item id.
+     */
+    readonly getDatabase: () => Promise<string>;
```

Comment:
I think this PR would be a good opportunity to refactor the API. For example - we discussed in standup how we don't want you parsing the tree id. Instead you can return an object like this (or whatever properties you need):
```
{
    id: string;
    accountName: string;
    databaseName: string;
}
```

Also couldn't you just include the connection string on that object and get rid of the `getConnectionString` function?

Reply:
If I don't parse id than I know 2 options of getting accountName & databaseName:
1. When CosmosDB node is revealing in AppService explorer under Connections node I need to run some sort of getDatabase API function to get those Names
2. Store this info in workspace settings

I feel like the first option is really bad, and I'm not quite sure that the idea of storing the additional info in workspace settings is good. I prefer to return some sort of "stable" id as @StephenWeatherford mentioned [here](https://github.com/Microsoft/vscode-cosmosdb/pull/886#discussion_r227990941) and parse it to get accountName & databaseName.

It's possible to get rid of the `getConnectionString`, but if I do then there is no way to get a connection string after I picked right database after getDatabase() call and got settings added. I don't like this because it complicates creating some sort of re-connect procedure that possibly could be useful:
- In the case of multiple connections of the same type, you could change which one is in the application settings(actually connected) without deleting/re-adding
- If any of appSetting were accidentally deleted you can restore it without deleting/re-adding

Haven't spoken about this with Matt, so absolutely not sure that it's the things that someday we're going to implement.

So, if we'll decide to delete getConnectionString method I can return object like this: 
```
{
    id: string,
    connectionString: string
}
```

Reply:
> When CosmosDB node(in AppService explorer under Connections node) is revealing I need to run some sort of getDatabase API function to get those Names

Are you talking about code in Cosmos or in App Service?  CosmosDB owns the format of the ID, it's perfectly fine for it to parse/interpret the value.

App Service should make no assumptions about the ID format.

Reply:
> App Service should make no assumptions about the ID format.

That's a reason why I need to create conversion function that you mentioned, so AppService makes an assumption about id from conversion function. Do I understand right?

That's how I store a connection in workspace settings:
![image](https://user-images.githubusercontent.com/15349723/47523294-8f69f480-d84c-11e8-9271-75ac70216e87.png)
accountName and databaseName I got from those strings under "appService.connections" -> "cosmosDB". I need them for UI:
![image](https://user-images.githubusercontent.com/15349723/47523536-233bc080-d84d-11e8-829e-95a2c461845f.png)


I'm going to implement conversion function, which will return always the same ID value for the node, so this "external" ID will be always the same and you don't need to change it in all extensions that uses it.

Reply:
Let me know when we can discuss.